jest/private/jest_test.bzl

```diff
@@ -79,3 +93,22 @@ lib = struct(
     attrs = _attrs,
     implementation = _impl,
 )
+
+def _jest_junit_reporter_path(ctx):
+    # Get the reporter path for jest-junit. Jest resolves reporters relative to the
+    # user config file, so we need the relative path from the config to the runfiles
+    # location of our internal dep on jest-junit.
+
+    short_path = ctx.file.config.short_path
+    if short_path.startswith("../"):
+        short_path = short_path[3:]
+
+    if paths.dirname(short_path) == "":
+        path_to_runfiles_root_from_config = "."
+    else:
+        path_to_runfiles_root_from_config = "/".join([".." for segment in paths.dirname(short_path).split("/")])
+
+    return "{path_to_runfiles_root_from_config}/external/{jest_repository}/node_modules/jest-junit/index.js".format(
```

Comment:
ohhh. I think this is still broken and depending on legacy external runfiles. you should set the `--nolegacy_external_runfiles` flag on the repository.

runfiles layout looks like this without the flag

```
something.runfiles/__main__/...
something.runfiles/{jest_repository}/...
something.runfiles/{other_repository}/...
```

The `short_path` of jest config in the main repository like so `something.runfiles/__main__/foo/bar/jest.config.js`: `foo/bar/jest.config.js`

The `short_path` of jest config in the main repository like so `something.runfiles/{other_repository}/foo/bar/jest.config.js`: `../{other_repository}/foo/bar/jest.config.js`

In both case you want to the number of `..` segments to be `1 + segments_in_package_path`.

So I think the general solution is:

```
path_to_root = "/".join([".."] * (1 + (len(ctx.file.config.label.package.split("/") if ctx.file.config.label.package else 0))
return return "{path_to_root}/{jest_repository}/node_modules/jest-junit/index.js".format(...)
```

Reply:
This didn't quite work. See my new solution which is much cleaner.

Reply:
nice